subsys/net/lib/download_client/src/download_client.c

```diff
@@ -462,7 +462,7 @@ static int request_resend(struct download_client *dl)
 		if (IS_ENABLED(CONFIG_COAP)) {
 			rc = coap_initiate_retransmission(dl);
 			if (rc) {
-				error_evt_send(dl, errno);
+				error_evt_send(dl, ECONNABORTED);
```

Comment:
This change is correct, as I don't see `coap_initiate_retransmission()` setting any errno values.

However, I think the correct return code would be `ETIMEDOUT` as we cannot really claim that someone aborted the connection.